### PR TITLE
Remove dependencies upon removing a job

### LINKED ISSUE #107: Removing a job doesn't remove related DependentJobRegistryEntry entries.
**Describe the bug**
See comment https://github.com/NCronJob-Dev/NCronJob/pull/106#discussion_r1820165053

### PR DESCRIPTION
**Pull request by nulltoken**
## Pull request description
Fix #107

### PR meta checklist
- [x] Pull request is targeted at `main` branch for code   
- [x] Pull request is linked to all related issues, if any.

### Code PR specific checklist
- [x] My code follows the code style of this project and AspNetCore coding guidelines.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [x] I have updated the appropriate sub section in the _CHANGELOG.md_.
- [x] I have added, updated or removed tests to according to my changes.
  - [x] All tests passed.

### CODE REVIEW

**Comment by linkdotnet** [comment]
Love it and will merge it. 

PS: Would it make things easier for you to be "Contributor" and be basically part of the NCronJob-Dev org?

**Comment by linkdotnet** [comment]
Maybe one nitpick: Can you add an entry to the changelog.md that explains your fix (plus giving you credit)? (Under the UNRELEASED section).

File: tests/NCronJob.Tests/RunDependentJobTests.cs [hunk 1]
```diff
@@ -36,6 +36,35 @@ public async Task WhenJobWasFailed_DependentJobShouldRun()
...
+
+        var result = await CommunicationChannel.Reader.ReadAsync(CancellationToken);
+        Assert.Equal(nameof(MainJob), result);
+        result = await CommunicationChannel.Reader.ReadAsync(CancellationToken);
+        Assert.Equal(nameof(SubMainJob), result);
+
+        var registry = provider.GetRequiredService<IRuntimeJobRegistry>();
+
+        registry.RemoveJob<PrincipalCorrelationIdJob>();
+        registry.AddJob(n => n.AddJob<MainJob>());
```
Comment (nulltoken) [inline tests/NCronJob.Tests/RunDependentJobTests.cs:58]: @linkdotnet FWIW, this syntax (`AddJob(AddJob)`) seems a bit weird.

Wouldn't there be, in the context of v4, an opportunity to find a less "redundant" naming?

Comment (linkdotnet) [reply tests/NCronJob.Tests/RunDependentJobTests.cs:58]: Yes - what I wanted is to reuse the same builder so that people only have to "learn" one API.

**Review by linkdotnet — APPROVED** [review]

**Comment by nulltoken** [comment]
> Can you add an entry to the changelog.md that explains your fix (plus giving you credit)? (Under the UNRELEASED section).

Done.

File: tests/NCronJob.Tests/RunDependentJobTests.cs (hunk 1 above)
Comment (nulltoken) [reply tests/NCronJob.Tests/RunDependentJobTests.cs:58]: Tracked through #122

**Comment by nulltoken** [comment]
> > Can you add an entry to the changelog.md that explains your fix (plus giving you credit)? (Under the UNRELEASED section).
> 
> Done.

I took the liberty to also cleanup the changelog file a bit:
- Target the `NCronJob-Dev` organization
- Fix tag names in compare links

**Comment by nulltoken** [comment]
> PS: Would it make things easier for you to be "Contributor" and be basically part of the NCronJob-Dev org?

That'd be nice. Thanks for the proposal!

**Comment by linkdotnet** [comment]
> > > Can you add an entry to the changelog.md that explains your fix (plus giving you credit)? (Under the UNRELEASED section).
> > 
> > 
> > Done.
> 
> I took the liberty to also cleanup the changelog file a bit:
> 
> * Target the `NCronJob-Dev` organization
> * Fix tag names in compare links

Thanks - Initially it was a deliberate choice to keep the old name, mainly:
1. Some of the releases (everything prior 2.5 or so) is not under the NCronJob namespace and package
2. I see the changelog as "append" only in the majority of cases.
But it also doesn't hurt either - so let's roll with it

**Review by linkdotnet — APPROVED** [review]

**Comment by linkdotnet** [comment]
Feel free to merge

**Pull request merged**